oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -787,7 +787,21 @@ private static QueryBuilder fullTextQuery(String text, String fieldName, PlanRes
             // and could contain other parts like renditions, node name, etc
             return multiMatchQuery.field(fieldName);
         } else {
-            return simpleQueryStringQuery(text).field(fieldName).defaultOperator(Operator.AND);
+            Boolean allowLeadingWildcard = false;
+            for (PropertyDefinition pd : pr.indexingRule.getProperties()) {
+                if (pd.name.equals(fieldName)) {
+                    allowLeadingWildcard = pd.allowLeadingWildcard;
+                }
+            }
+            // https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html
+            // simpleQueryStringQuery does not support leading wildcards whereas it's supported by default in queryStringQuery
+            // Setting this to true can have performance impact, hence it's false by default in oak implementation for elastic,
+            // However clients can set it to true from property definition based on case by case usage.
+            if (allowLeadingWildcard) {
+                return queryStringQuery(text).field(fieldName).defaultOperator(Operator.AND);
+            } else {
+                return simpleQueryStringQuery(text).analyzeWildcard(true).field(fieldName).defaultOperator(Operator.AND);
+            }
```

Comment:
@fabriziofortino  - yes, usage for this is not documented on lucene and I suppose it was found out by SITES team while working on https://jira.corp.adobe.com/browse/SITES-2564.

I added this check for specific properties/fields because elastic docs mention that this can have significant performance impact, so I didn't want to give an option to enable this at index level.

>Does it mean that for full-text queries (eg: contains(. ,"*foo*") this would never work?
atm, yes..we can change this behaviour, but I am not sure if we should. (I have not tested what the corresponding behaviour in lucene), but imo, we should restrict this to per field basis. wdyt ?